pipeline_steps/common.groovy

```diff
@@ -1513,13 +1513,61 @@ def findHookDir(String hook_dirs) {
 }
 
 void runReleasesPullRequestWorkflow(String baseBranch, String prBranch, String jiraProjectKey){
-  (prType, componentText) = getComponentChange(baseBranch, prBranch)
+  def (prType, componentText) = getComponentChange(baseBranch, prBranch)
+  def componentYaml = readYaml text: componentText
 
   if (prType == "release"){
-    testRelease(component_text)
-    createRelease()
+    String REPO_NAME=componentYaml["name"]
+    String REPO_URL=componentYaml["repo_url"]
+    String SHA=componentYaml["release"]["sha"]
+    String MAINLINE=componentYaml["release"]["series"]
+    String RC_BRANCH="${MAINLINE}-rc"
+
+    dir(REPO_NAME){
+      clone_with_pr_refs(
+        "./",
+        REPO_URL,
+        "master",
+      )
+      println "=== Checking that the specified SHA exists ==="
+      Integer sha_exists = ! sh(
+        script: """#!/bin/bash -xe
+          git rev-parse --verify ${SHA}^{commit}
+        """,
+        returnStatus: true,
+      )
+      if (! sha_exists) {
+        throw new Exception("The supplied SHA ${SHA} was not found.")
+      }
+      println "=== Checking for the existence of an RC branch ==="
+      Integer has_rc_branch = ! sh(
```

Comment:
This needs to be Boolean

https://groovyconsole.appspot.com/script/5089364245741568

Reply:
Fixed!